custom/ibc-transfer/keeper/msg_server.go

```diff
@@ -30,8 +31,17 @@ func (k msgServer) Transfer(goCtx context.Context, msg *types.MsgTransfer) (*typ
 	if params.ChannelFees != nil && len(params.ChannelFees) > 0 {
 		channelFee := findChannelParams(params.ChannelFees, msg.SourceChannel)
 		if channelFee != nil {
-			if channelFee.MinTimeoutTimestamp > 0 && msg.TimeoutTimestamp < channelFee.MinTimeoutTimestamp {
-				return nil, fmt.Errorf("incorrect timeout timestamp found during ibc transfer")
+			if channelFee.MinTimeoutTimestamp > 0 {
+
+				goCtx := sdk.UnwrapSDKContext(goCtx)
+				blockTime := goCtx.BlockTime()
+
+				timeoutTimeInFuture := time.Unix(0, int64(msg.TimeoutTimestamp))
+				difference := timeoutTimeInFuture.Sub(blockTime).Nanoseconds()
```

Comment:
What if `timeoutTimeInFuture < blockTime`?

Reply:
added extra validation. thanks.

if timeoutTimeInFuture.Before(blockTime) {
					return nil, fmt.Errorf("incorrect timeout timestamp found during ibc transfer. timeout timestamp is in the past")
				}